pkg/signup/signup_service.go

```diff
@@ -68,15 +117,54 @@ func (c *SignupServiceImpl) CreateUserSignup(username, userID string) (*crtapi.U
 		},
 	}
 
-	created, err := c.UserSignups.Create(userSignup)
+	created, err := s.UserSignups.Create(userSignup)
 	if err != nil {
 		return nil, err
 	}
 
 	return created, nil
 }
 
-func (c *SignupServiceImpl) transformAndValidateUserName(username string) (string, error) {
+// getUserSignupImpl gets the UserSignup resource with the specified userID
+// Returns nil, nil if the resource is not found
+func (s *ServiceImpl) getUserSignupImpl(userID string) (*Signup, error) {
+	// get signup resource
+	userSignup, err := s.UserSignups.Get(userID)
+	if err != nil {
+		return nil, err
+	}
+	// get MUR for it
+	mur, err := s.MasterUserRecords.Get(userSignup.GetName())
```

Comment:
It will be a bit more complex than that. userSignup.Name will hold userID. And MUR.Name will hold the transformed username.
So, if:
```
original RHD IdentityProvider (IDP) username: jsmith@gmail.com
transformed Kube username: jsmith_at_gmail.com
RHD userID: 12345
```
Then:
```
UserSignup.Name: 12345
UserSignup.Spec.Username: jsmith_at_gmail.com
UserSignup.Meta.Annotations["idp_username"]: jsmith@gmail.com

MUR.Name: jsmith_at_gmail.com
MUR.Spec.UserID: 12345
MUR.Meta.Annotations["idp_username"]: jsmith@gmail.com
```

Reply:
@sbryzak can you take a look here and fix this?